crates/ide-assists/src/handlers/generate_derive.rs

```diff
@@ -42,7 +42,8 @@ pub(crate) fn generate_derive(acc: &mut Assists, ctx: &AssistContext<'_>) -> Opt
                 .next();
             match derive_attr {
                 None => {
-                    builder.insert_snippet(cap, node_start, "#[derive($0)]\n");
+                    let indent_level = IndentLevel::from_node(nominal.syntax());
+                    builder.insert_snippet(cap, node_start, format!("#[derive($0)]\n{indent_level}"));
```

Comment:
**NB** If there's a comment, it will be considered a part of `nominal: Adt`, so `indent_level` will be the comment's indent. If this comment and the line containing ADT name have different indents, the indent of later will be changed.
Thus it might be better to calculate here the indent of the ADT name. On the other hand, this assist is expected only to _preserve_ correct indentation, not to fix incorrect one.